automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/monitor/SubmissionDetailsPage.scala

```
@@ -79,13 +79,15 @@ class SubmissionDetailsPage(namespace: String, name: String, var submissionId: S
     * @param timeOut: Time out. Default set 20.minutes
     */
   def waitUntilSubmissionCompletes(timeOut: FiniteDuration = 20.minutes): Unit = {
+    logger.info("Waiting for Submission to complete")
     Thread.sleep(10000) // 10 seconds pause before checking
     retry[Boolean](10.seconds, timeOut) ({
       // sometimes page auto reloads, displaying table "Workflow Detail". link submissionId is not in table "Workflow Detail".
       // click tab "Monitor" loads either table "Workflow Details" or table "Analysis Detail".
       if (workflowStatusLabel.isVisible) {
         val monitorTab = goToMonitorTab()
         monitorTab.openSubmission(submissionId) // link exists in "Analysis Detail" table
+        await ready (submissionStatusLabel, 10)
```

defensive code: if `openSubmission` fails, this fails next. Instead of keep going and fails with false exception.

---

Can `openSubmission` do this itself? Navigation functions should only return once the navigation completes, or should throw an error on failure.

---

Label `submissionStatusLabel` is visible only in table `Workflow Details` and exist in this Page Object file only. When I have more free time in future, I could do some code refactoring and cleanup.